src/fetchClient.ts

```diff
@@ -53,25 +53,53 @@ export class FetchClient {
     };
   }
 
-  private static async returnData<T>(
+  private static async buildResponseDataObject<BodyType>(
     response: Response,
-    sanitizer?: DataSanitizer<T>,
-  ): ResponseType<object, T> {
-    const text = await response.text();
-    const data = text.length > 0 ? JSON.parse(text) : {};
+    body: BodyType,
+  ) {
     const { headers, ok, redirected, status, statusText, type, url } = response;
+
     return {
       headers,
-      ok,
       redirected,
       status,
       statusText,
       type,
       url,
-      body: sanitizer ? sanitizer(data) : data,
+      body,
+      ok,
     };
   }
 
+  private static async returnData<BodyType>(
+    response: Response,
+    sanitizer?: DataSanitizer<BodyType>,
+  ): ResponseType<object, BodyType> {
+    const text = await response.text();
+    if (text.length === 0) {
+      return FetchClient.buildResponseDataObject(
+        response,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        {} as unknown as any,
+      );
+    }
+
+    try {
+      const parsedBody = JSON.parse(text);
+      return FetchClient.buildResponseDataObject(
+        response,
+        sanitizer ? sanitizer(parsedBody) : parsedBody,
+      );
+    } catch (_error) {
```

Comment:
We could match the explicit JSON parse error before we rethrow and rethrow original if it's not that